### PR TITLE
[Issue 3307] [feature] Extension - Allow to use Migration Polkadot Appp to attach Ledger account

### PR DESCRIPTION
**Pull request by Thiendekaco**

### CODE REVIEW

File: packages/extension-koni-ui/src/Popup/Confirmations/parts/Sign/Substrate.tsx [hunk 1]
```diff
@@ -254,7 +254,7 @@ const Component: React.FC<Props> = (props: Props) => {
     refresh: refreshLedger,
     signMessage: ledgerSignMessage,
     signTransaction: ledgerSignTransaction,
-    warning: ledgerWarning } = useLedger(chainSlug, activeLedger, true, isRuntimeUpdated || isMessage);
+    warning: ledgerWarning } = useLedger(chainSlug, activeLedger, true, isRuntimeUpdated || isMessage || !!account.slip44, account.slip44);
```
Comment (S2kael) [inline packages/extension-koni-ui/src/Popup/Confirmations/parts/Sign/Substrate.tsx:257]: Remove `!!account.slip44` condition

**Review by S2kael — CHANGES_REQUESTED** [review]

File: packages/extension-koni-ui/src/connector/Ledger/SubstrateGenericLedger.ts [hunk 2]
```diff
@@ -46,8 +47,8 @@ export class SubstrateGenericLedger extends BaseLedger<PolkadotGenericApp> {
   getAddress (confirm?: boolean, accountOffset?: number, addressOffset?: number, accountOptions?: Partial<AccountOptions>): Promise<LedgerAddress> {
     return this.withApp(async (app): Promise<LedgerAddress> => {
       const path = this.serializePath(accountOffset, addressOffset, accountOptions);
-
-      const { address, pubKey } = await this.wrapError(app.getAddress(path, 42, confirm));
+      const def = PredefinedMigrationLedgerNetwork.find(({ slip44 }) => slip44 === this.slip44);
+      const { address, pubKey } = await this.wrapError(app.getAddress(path, def?.ss58_addr_type || 42, confirm));
```
Comment (S2kael) [inline packages/extension-koni-ui/src/connector/Ledger/SubstrateGenericLedger.ts:51]: Replace `||` by `??`

**Review by S2kael — CHANGES_REQUESTED** [review]

**Comment by saltict** [comment]
🚀 Deployed on https://pr-3346--sw-web-runner.netlify.app

File: packages/extension-koni-ui/src/hooks/ledger/useLedger.ts [hunk 3]
```diff
@@ -46,20 +46,32 @@ const getNetwork = (ledgerChains: LedgerNetwork[], slug: string, isEthereumNetwo
   return ledgerChains.find((network) => network.slug === slug || (network.isEthereum && isEthereumNetwork));
 };
 
-const retrieveLedger = (slug: string, ledgerChains: LedgerNetwork[], isEthereumNetwork: boolean, forceMigration: boolean): Ledger => {
+const getNetworkBySlug = (ledgerChains: MigrationLedgerNetwork[], slug: string): MigrationLedgerNetwork | undefined => {
+  return ledgerChains.find((network) => network.slug === slug);
+};
+
+const retrieveLedger = (slugApp: string, ledgerChains: LedgerNetwork[], migrateLedgerChains: MigrationLedgerNetwork[], isEthereumNetwork: boolean, forceMigration: boolean, slugNetwork?: string): Ledger => {
```
Comment (S2kael) [inline packages/extension-koni-ui/src/hooks/ledger/useLedger.ts:53]: `chainSlug` not `slugApp`

File: packages/extension-koni-ui/src/constants/ledger.ts [hunk 4]
```diff
@@ -35,45 +49,6 @@ export const PredefinedLedgerNetwork: LedgerNetwork[] = [
...
-    genesisHash: ChainInfoMap.statemine.substrateInfo?.genesisHash || '0x48239ef607d7928874027a43a67689209727dfb3d3dc5e5b03a39bdc2eda771a',
-    icon: 'substrate',
-    network: 'statemine',
-    slug: ChainInfoMap.statemine.slug,
-    isDevMode: false,
-    isGeneric: false,
-    isEthereum: false,
-    slip44: 434
-  },
   {
```
Comment (S2kael) [inline packages/extension-koni-ui/src/constants/ledger.ts:52]: Sorry, but please rollbacks change, hide option instead remove

File: packages/extension-koni-ui/src/constants/ledger.ts [hunk 5]
```diff
@@ -235,12 +210,135 @@ export const PredefinedLedgerNetwork: LedgerNetwork[] = [
   // }
 ];
 
+export const PredefinedMigrationLedgerNetwork: MigrationLedgerNetwork[] = [
```
Comment (S2kael) [inline packages/extension-koni-ui/src/constants/ledger.ts:213]: Please update this list

**Review by S2kael — APPROVED** [review]

File: packages/extension-koni-ui/src/Popup/Confirmations/index.tsx [hunk 6]
```diff
@@ -80,9 +79,16 @@ const Component = function ({ className }: Props) {
               canSign = true;
             } else {
               const payload = request.request.payload as SignerPayloadJSON;
+              let isRuntimeUpdated: boolean;
+
+              if (isRawPayload(payload)) {
```
Comment (S2kael) [inline packages/extension-koni-ui/src/Popup/Confirmations/index.tsx:84]: No need to check if `payload` is a message, this condition is checked before

File: packages/extension-koni-ui/src/hooks/ledger/useLedger.ts [hunk 7]
```diff
@@ -46,20 +46,38 @@ const getNetwork = (ledgerChains: LedgerNetwork[], slug: string, isEthereumNetwo
   return ledgerChains.find((network) => network.slug === slug || (network.isEthereum && isEthereumNetwork));
 };
 
-const retrieveLedger = (slug: string, ledgerChains: LedgerNetwork[], isEthereumNetwork: boolean, forceMigration: boolean): Ledger => {
+const getNetworkByGenesisHash = (ledgerChains: MigrationLedgerNetwork[], genesisHash: string): MigrationLedgerNetwork | undefined => {
+  return ledgerChains.find((network) => network.genesisHash === genesisHash);
+};
+
+const retrieveLedger = (chainSlug: string, ledgerChains: LedgerNetwork[], migrateLedgerChains: MigrationLedgerNetwork[], isEthereumNetwork: boolean, forceMigration: boolean, slugNetwork?: string | null): Ledger => {
```
Comment (S2kael) [inline packages/extension-koni-ui/src/hooks/ledger/useLedger.ts:53]: Change name of `slugNetwork`

**Pull request merged**